src/stream/src/executor/hash_agg.rs

```diff
@@ -540,14 +676,16 @@ impl<K: HashKey, S: StateStore> HashAggExecutor<K, S> {
         for msg in input {
             let msg = msg?;
             match msg {
-                Message::Watermark(mut watermark) => {
+                Message::Watermark(watermark) => {
                     let group_key_seq = group_key_invert_idx[watermark.col_idx];
                     if let Some(group_key_seq) = group_key_seq {
-                        watermark.col_idx = group_key_seq;
-                        vars.buffered_watermarks[group_key_seq] = Some(watermark);
+                        if group_key_seq == 0 {
```

Comment:
Can we make sure that window watermark is on the first group key?

Reply:
Yes we assume window column is the first group key. This will be guarranteed by frontend.